lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -230,6 +246,27 @@ public function execute($params = null)
         return true;
     }
 
+    /**
+     * Send BLOB type streams using mysqli::send_long_data
+     */
+    private function _sendLongData()
+    {
+        foreach ($this->_longData as $paramNr => $resource) {
+            if (is_resource($resource) && get_resource_type($resource) === 'stream') {
```

Comment:
Should we throw an exception if it's a non-stream resource? No point in trying to send the resource itself in the `else` block.

Reply:
It can be either a resource or a string. I'll rename the variable.

Reply:
:+1: